modules/common/tls/tls.go

```diff
@@ -435,3 +443,17 @@ func (c *Ca) CreateVolume() corev1.Volume {
 
 	return volume
 }
+
+// GetAdditionalSubjectNames represents a utility function that takes a comma
+// separated list of additionalSubjectNames and returns a list of string that
+// can be easily handled by openstack-operator when a CertificateRequest is
+// generated.
+func GetAdditionalSubjectNames(svc corev1.Service) []string {
```

Comment:
how about we add a generic func to https://github.com/openstack-k8s-operators/lib-common/blob/main/modules/common/util/map.go where we pass in a map and the key to lookup from the hash . then the bellow and return the list. I think it might be  useful for other cases

Reply:
ok, it's something I had in my initial version, but then I thought it was better to have a `Service` as input to hide details at the openstack-operator level.
I'll change this function to be something like:

```golang
// GetStringsFromMap - It returns a list of string based on a comma separated
// list of strings. This is usually set to normalize annotations
func GetStringsFromMap(hash map[string]string, key string) []string {
	res := []string{}
	if h, ok := hash[key]; ok {
		if h != "" {
			res = strings.Split(h, ",")
		}
	}
	return res
}
```

and add some unit tests to it.
Let me know if it's not something that matches with what you had in mind.